js/util/combineCriteria.ts

```diff
@@ -0,0 +1,31 @@
+import * as Collections from 'typescript-collections';
+import { PropExtractorCriteria, PropExtractorConfig } from './extractProps';
+
+export const getCombinedCriteria = (
+  componentName: string,
+  configs: PropExtractorConfig[]
+): PropExtractorCriteria => {
+  let remainingConfigs: PropExtractorConfig[] = [...configs];
```

Comment:
const

Reply:
Not as written, since it gets modified in the loop, but to be honest, I don't know why I took that approach, and it's kinda ugly.  Changing it to a for loop.

Reply:
Not actually modified - it's mutated with `.shift()`, but its value is never reassigned.